opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/servlet/SpanWrapper.java

```diff
@@ -0,0 +1,45 @@
+package io.opentracing.contrib.spring.web.servlet;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.opentracing.Span;
+
+/**
+ * Wrapper class used for exchanging span between filters
+ * It can be accessed via {@link javax.servlet.http.HttpServletRequest#getAttribute(String)} with name
+ * {@link TracingHandlerInterceptor#ACTIVE_SPAN}.
+ *
+ * @author Pavol Loffay
+ */
+public class SpanWrapper {
+
+    private Span span;
+    private AtomicBoolean finished = new AtomicBoolean();
+    private boolean spanFromFilter;
+
+    public SpanWrapper(Span span, boolean spanFromFilter) {
+        this.span = span;
+        this.spanFromFilter = spanFromFilter;
+    }
+
+    public Span get() {
+        return span;
+    }
+
+    /**
+     * Finish is idempotent
+     */
+    public synchronized void finish() {
+        if (!finished.get()) {
+            finished.set(true);
```

Comment:
You can probably get rid of the synchronized on the method if you use the `getAndSet` from the `AtomicBoolean`. Something like this:

```java
public void finish() {
    boolean wasFinished = finished.getAndSet(true); // this returns the previous value of finished and is atomic
    if (!wasFinished && !spanFromFilter) {
      span.finish();
    }
}
```